src/spaces/grid.jl

```diff
@@ -16,15 +16,16 @@ struct Hood{D} # type P stands for Periodic and is a boolean
     βs::Vector{CartesianIndex{D}} # neighborhood cartesian indices
 end
 
-struct GridSpace{D,P} <: DiscreteSpace
+struct GridSpace{D,P,W<:PathfinderType} <: DiscreteSpace
     s::Array{Vector{Int},D}
+    pathfinder::W
```

Comment:
Please add new fields to the **last position** of an existing Type decleration. No reason to go and alter the ordering of all existing fields in the code.